Modules/_pickle.c

```diff
@@ -1789,6 +1789,7 @@ fast_save_enter(PicklerObject *self, PyObject *obj)
             return 0;
         }
         if (PyErr_Occurred()) {
+            Py_DECREF(key);
```

Comment:
I think it is worth to add `self->fast_nesting = -1;` here and before other `return 0` above.

Reply:
Done!